src/api/store/community.py

```diff
@@ -1,16 +1,206 @@
 from _main_.utils.massenergize_response import MassenergizeResponse
 from _main_.utils.context import Context
 from django.db.models import Q
-from .utils import get_community_or_die, get_user_or_die, get_new_title
-import random 
+from .utils import get_community_or_die, get_user_or_die, get_new_title, is_reu_in_community
+import random
+import zipcodes
 from sentry_sdk import capture_message, capture_exception
 
 class CommunityStore:
   def __init__(self):
     self.name = "Community Store/DB"
 
+  # QUESTION FOR SAM:
+  # These helper routines don't make use of self, should they be member functions as currently implemented, 
+  # or outside the CommunityStore class?
+
+  def _check_geography_unique(self, community, geography_type, loc):
+    # ensure that this zip code is not part of another geographic zip code based community
+    check_communities = Community.objects.filter(is_geographically_focused=True, geography_type=geography_type, is_deleted=False).prefetch_related('location_set')
+    for check_community in check_communities:
+      if check_community.id == community.id:
+        continue
+      for location in check_community.location_set.all():
+        if geography_type == 'ZIPCODE' and location.zipcode == loc:
+          # zip code already used
+          message = 'Zipcode %s is already part of another geographic community %s.' % (loc, check_community.name)
+          print(message)
+          raise Exception(message)
+        elif geography_type == 'CITY' and location.city == loc:
+          message = 'City %s is already part of another geographic community %s.' % (loc, check_community.name)
+          print(message)
+          raise Exception(message)
+        elif geography_type == 'COUNTY' and location.county == loc:
+          message = 'County %s is already part of another geographic community %s.' % (loc, check_community.name)
+          print(message)
+          raise Exception(message)
+        elif geography_type == 'STATE' and location.state == loc:
+          message = 'State (%s) is already part of another geographic community (%s).' % (loc, check_community.name)
+          print(message)
+          raise Exception(message)
+        elif geography_type == 'COUNTRY' and location.country == loc:
+          message = 'Country (%s) is already part of another geographic community (%s).' % (loc, check_community.name)
+          print(message)
+          raise Exception(message)
+
+  def _update_locations(self, geography_type, location_set, community):
+        
+    # clean up in case there is garbage in there
+    if community.location_set:
+      community.location_set.clear()
+
+    # this is a list of zipcodes, towns, cities, counties, states        
+    location_list = location_set.replace(", ",",").split(",")  # passed as comma separated list
+    print("Community includes the following locations :"+str(location_list))
```

Comment:
remove print statement

Reply:
The print statements are important for me to validate that nothing goes wrong in the deployment to databases.  I don't think they do any harm but if you insist I will go back and eliminate them once the databases are migrated and the community definitions set.